Tests/SwiftDocCTests/LinkTargets/LinkDestinationSummaryTests.swift

```diff
@@ -829,4 +829,51 @@ class LinkDestinationSummaryTests: XCTestCase {
             "doc://com.example.mymodule/documentation/MyModule/MyClass/myFunc()-9a7po",
         ])
     }
+
+    /// Tests that API Collections (articles with Topics sections) are correctly identified as `.collectionGroup`
+    /// kind in linkable entities, ensuring cross-framework references display the correct icon.
+    func testAPICollectionKindForLinkDestinationSummary() async throws {
+        let symbolGraph = makeSymbolGraph(
+            moduleName: "TestModule",
+            symbols: [makeSymbol(id: "test-class", kind: .class, pathComponents: ["TestClass"])]
+        )
+
+        let catalogHierarchy = Folder(name: "unit-test.docc", content: [
+            TextFile(name: "APICollection.md", utf8Content: """
+                # Time Pitch Algorithm Settings
+
+                This is an API Collection that curates symbols.
+
+                ## Topics
+
+                ### Algorithms
```

Comment:
This topic name doesn't match the symbol that it's curating.

Because the API collection only has a single topic section, you can probably remove its title if we don't have a better title for it.
```suggestion
```